MaterialLib/MPL/Properties/RelPermVanGenuchten.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "RelPermVanGenuchten.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include "MaterialLib/MPL/Medium.h"
+
+namespace MaterialPropertyLib
+{
+RelPermVanGenuchten::RelPermVanGenuchten(
+    double const residual_liquid_saturation,
+    double const residual_gas_saturation,
+    double const min_relative_permeability_liquid,
+    double const exponent)
+    : _S_L_res(residual_liquid_saturation),
+      _S_L_max(1. - residual_gas_saturation),
+      _k_rel_min(min_relative_permeability_liquid),
+      _m(exponent)
+{
+    if (!(_m > 0 && _m < 1))
+    {
+        OGS_FATAL(
+            "The exponent parameter of vanGenuchten, m=%f is out of its range "
+            "of (0, 1)",
+            _m);
+    }
+}
```

Comment:
Sorry, I see there is one similar guard in  createRelPermVanGenuchten. It is better to keep it in createRelPermVanGenuchten. The same for createSaturationVanGenuchten. The new change in SaturationVanGenuchten::SaturationVanGenuchten can be moved to createSaturationVanGenuchten.

Reply:
I see. The constructor is good place though, because not all of the objects are created through the create functions, *e.g.* the tests.